### PR TITLE
cmd/contour: adds sds cli

### PR DESCRIPTION
**Pull request by mattalberts**
Addresses the third high-level design bullet point, support sds in the
contour cli for debugging.
    
Updates: #898
Signed-off-by: Matt Alberts <malberts@cloudflare.com>

### CODE REVIEW

**Comment by mattalberts** [comment]
The review and commit order would be
 1. #1046 
 1. #1047 
 1. #1048

File: internal/dag/dag.go [hunk 1]
```diff
@@ -293,6 +293,16 @@ func (s *Secret) Data() map[string][]byte {
 	return s.Object.Data
 }
 
+// Cert returns the secret's tls certificate
+func (s *Secret) Cert() []byte {
+	return s.Object.Data[v1.TLSCertKey]
```
Comment (stevesloka) [inline internal/dag/dag.go]: Could there be a case where the cert is not in the secret?

File: internal/dag/dag.go [hunk 2]
```diff
@@ -293,6 +293,16 @@ func (s *Secret) Data() map[string][]byte {
...
+}
+
+// PrivateKey returns the secret's tls private key
+func (s *Secret) PrivateKey() []byte {
+	return s.Object.Data[v1.TLSPrivateKeyKey]
```
Comment (stevesloka) [inline internal/dag/dag.go]: Same here if the key is nil?

File: internal/dag/dag.go (hunk 1 above)
Comment (mattalberts) [reply internal/dag/dag.go]: It's plausible.  It is assumed to be valued when setting up the downstream context; but on review, that isn't guaranteed. 
 - `envoy.DownstreamTLSContext(data[v1.TLSCertKey], data[v1.TLSPrivateKeyKey], vh.MinProtoVersion, alpnProtos...)`
 - https://github.com/heptio/contour/blob/master/internal/contour/listener.go#L265

It feels like the the structure should be verified while creating a `SecureVirtualHost` vertex.

File: internal/dag/dag.go (hunk 2 above)
Comment (mattalberts) [reply internal/dag/dag.go]: Agreed. if the key doesn't exist, the return would be populated as and empty `[]byte`.

File: internal/dag/dag.go (hunk 1 above)
Comment (mattalberts) [reply internal/dag/dag.go]: These got marked as outdated when I rebased. They were brought in with #1047.

Comment (stevesloka) [reply internal/dag/dag.go]: ok no biggie, there's another issue where we get the cert and not the key and it causes issues, but let's not try and fix that in this (just been on my mind).

File: internal/dag/dag.go (hunk 2 above)
Comment (stevesloka) [reply internal/dag/dag.go]: :+1:

File: internal/dag/dag.go (hunk 1 above)
Comment (mattalberts) [reply internal/dag/dag.go]: Its okay! I'm glad you mentioned it.  I remember looking at it while I drafted the PR, when trying to decide on introducing the helper functions on *dag.Secret.  Is there a already an issue for that?  I can clean them all up at one time in another PR :).

Comment (mattalberts) [reply internal/dag/dag.go]: I bet its this one, #1051  :)

**Pull request merged**